tdp/core/dag.py

```diff
@@ -46,21 +46,25 @@
 }
 DEFAULT_SERVICE_PRIORITY = 99
 
+DAG_FOLDER_NAME = "tdp_lib_dag"
+DAG_EXTENSION = ".yml"
+
 
 class Dag:
     """Generate DAG with components dependencies"""
 
-    def __init__(self, yaml_files=None, playbooks_dir=None):
+    def __init__(self, yaml_files, playbooks_dir=None):
         self._components = None
         self._graph = None
         self._yaml_files = None
         self._services = None
         self._services_components = None
+        self.yaml_files = yaml_files
         self.playbooks_dir = playbooks_dir
 
-        if yaml_files is None:
-            yaml_files = list((Path(tdp.components.__file__).parent).glob("*.yml"))
-        self.yaml_files = yaml_files
+    @staticmethod
+    def from_collection(collection_path):
+        return Dag((collection_path / DAG_FOLDER_NAME).glob("*" + DAG_EXTENSION))
```

Comment:
What do you think about using `Path` constructor in order to accept `str` for `collection_path` parameter ? Actually, `collection_path` must be a `Path` object. If `collection_path` is a `str`, the `/` operator will not work.

Reply:
done in 09bd0e0